controllers/events.js

```diff
@@ -86,7 +94,7 @@ const getAllEvents = async (req, res) => {
  * @param {Object} req - The Express request object.
  * @param {Object} res - The Express response object.
  * @returns {Object} An object containing the generated token and a success message in JSON format.
- * @throws {Error} If an error occurs while generating the token.
+ * @returns {Object} If an error occurs while generating the token.
  */
 const joinEvent = async (req, res) => {
```

Comment:
function is not checking the role property of the req.body object. This means that the function will allow users to join events with any role, even if they are not supposed to have access to the event.

is my understanding correct..?